containers.sh

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+images="${PWD}/docker/images"
+
+log(){
+  echo "$(date +'%T') $1"
+}
+
+networks(){
+  docker network create \
+    todo-app
+}
+
+volumes(){
+  docker volume create \
+    todo-mysql-data
+}
+
+todo_app(){
+  docker run \
+    --detach \
+    --publish 3000:3000 \
+    --workdir "/app" \
+    --volume "${images}"/getting-started/app:/app \
+    --network todo-app \
+    --env MYSQL_HOST=mysql \
+    --env MYSQL_USER=root \
+    --env MYSQL_PASSWORD=secret \
+    --env MYSQL_DB=todos \
+    node:12-alpine \
```

Comment:
Node 12 is actually EOL, I think you need at least 14 to get to a supported release: https://github.com/nodejs/Release#release-schedule